src/storage-controller/src/lib.rs

```diff
@@ -474,14 +475,28 @@ where
             location,
             grpc_client: self.config.parameters.grpc_client.clone(),
         };
+        for id in instance.active_ingestions() {
+            self.collections
+                .get_mut(id)
+                .expect("instance contains unknown ingestion")
+                .active_copies += 1;
+        }
         instance.add_replica(replica_id, config);
     }
 
     fn drop_replica(&mut self, instance_id: StorageInstanceId, replica_id: ReplicaId) {
-        self.instances
+        let instance = self
+            .instances
             .get_mut(&instance_id)
-            .unwrap_or_else(|| panic!("instance {instance_id} does not exist"))
-            .drop_replica(replica_id);
+            .unwrap_or_else(|| panic!("instance {instance_id} does not exist"));
+
+        for id in instance.active_ingestions() {
+            self.collections
+                .get_mut(id)
+                .expect("instance contains unknown ingestion")
+                .active_copies -= 1;
```

Comment:
We might have to remove collection state if this value drops to zero. For example, consider the case where a source is dropped but before we receive the `DroppedId` response from the replica we drop the replica. With the current code I think we would just leak the `CollectionState` for the dropped collection.

This is tricky because we can't always drop a collection when this counter goes to zero. All replicas in a cluster can be dropped without the sources on the same cluster being dropped as well.